### PR TITLE
feat(typescript_indexer): add xrefs for declarations in objects

### PR DESCRIPTION
**Pull request by ayazhafiz**
Shorthand property assignments like foo in a = {foo,} were
previously not indexed at all. This emits edges for shorthand
assignments similar to other property assignments. Also adds xrefs for
computed properties in objects and ignores lvalue literals in a
declaration.

_edit_: new description

### CODE REVIEW

**Comment by ayazhafiz** [comment]
cc @evmar

**Comment by ayazhafiz** [comment]
> It's not clear whether the `foo` on L3 should xref to the `foo` on L2 or the `foo` on L4. If you write it out as `foo: foo` then the two things to anchor to are obvious.

I was just thinking about this. Turns out with the current TypeScript AST it is impossible to determine the xref to the declaration a shorthand property expression takes the value of, at least from my playing on https://ts-ast-viewer.com/

File: kythe/typescript/indexer.ts [hunk 1]
```diff
@@ -984,10 +985,15 @@ class Visitor {
           this.visit(element);
         }
         break;
+      case ts.SyntaxKind.ComputedPropertyName:
+        this.visit(decl.name.expression);
+        break;
+      // Nothing meaningful can be indexed about a literal expression.
+      case ts.SyntaxKind.StringLiteral:
+      case ts.SyntaxKind.NumericLiteral:
```
Comment (evmar) [inline kythe/typescript/indexer.ts]: Maybe you intended these for another change (?)

File: kythe/typescript/testdata/object.ts [hunk 2]
```diff
@@ -1,14 +1,25 @@
...
+  [computed]: 0,
+
   //- @method defines/binding Method
   //- Method.node/kind function
-  method() {
-  }
+  method() {}
 };
 
 //- @property ref Property
```
Comment (evmar) [inline kythe/typescript/testdata/object.ts]: Maybe some test of accessing shortProperty here too?

**Comment by ayazhafiz** [comment]
This should clean up most `todo`s recorded when running the indexer.

File: kythe/typescript/indexer.ts (hunk 1 above)
Comment (ayazhafiz) [reply kythe/typescript/indexer.ts]: Sure, I'll put this in another change.

File: kythe/typescript/testdata/object.ts (hunk 2 above)
Comment (ayazhafiz) [reply kythe/typescript/testdata/object.ts]: Done. Also added xrefs to computed properties.

File: kythe/typescript/indexer.ts [hunk 3]
```diff
@@ -977,6 +979,10 @@ class Visitor {
         this.emitNode(vname, 'variable');
 
         this.emitEdge(this.newAnchor(decl.name), 'defines/binding', vname);
+
+        if (ts.isComputedPropertyName(decl.name)) {
+          this.visit(decl.name.expression);
+        }
```
Comment (evmar) [inline kythe/typescript/indexer.ts]: Maybe we should just recurse with ts.forEachChild?  Not sure.

Comment (ayazhafiz) [reply kythe/typescript/indexer.ts]: Done.

**Comment by ayazhafiz** [comment]
@kythe could you review?

**Pull request merged**